### PR TITLE
HTTP server request events wrong metrics events for pipelined requests

### PR DESCRIPTION
**Pull request by vietj**

### CODE REVIEW

File: src/main/java/io/vertx/core/http/impl/Http1xServerRequest.java [hunk 1]
```diff
@@ -558,6 +566,25 @@ private void onEnd() {
     }
   }
 
+  private void reportRequestComplete(Http1xServerRequest request) {
```
Comment (TFyre) [inline src/main/java/io/vertx/core/http/impl/Http1xServerRequest.java:569]: No need for parameter anymore

File: src/main/java/io/vertx/core/http/impl/Http1xServerRequest.java [hunk 2]
```diff
@@ -558,6 +566,25 @@ private void onEnd() {
...
+    HttpServerMetrics metrics = conn.metrics;
+    if (metrics != null) {
+      metrics.requestEnd(request.metric(), request, request.bytesRead());
+      conn.flushBytesRead();
+    }
+  }
+
+  private void reportRequestBegin(Http1xServerRequest request) {
```
Comment (TFyre) [inline src/main/java/io/vertx/core/http/impl/Http1xServerRequest.java:577]: No need for parameter anymore

**Comment by vietj** [comment]
right thanks!

File: src/main/java/io/vertx/core/http/impl/Http1xServerRequest.java (hunk 2 above)
Comment (vietj) [reply src/main/java/io/vertx/core/http/impl/Http1xServerRequest.java]: thanks I updated

**Comment by TFyre** [comment]
The only other question:
https://github.com/eclipse-vertx/vert.x/pull/4088#issuecomment-917979726
Should it not calculate the total time for the request (including time spent waiting in the queue)?

**Comment by vietj** [comment]
well in practice pipelining should not be used, so most of the time it
will be correct

**Comment by vietj** [comment]
that being said we could indeed initialize it maybe all the time when
the request has been received ?

in that case the metrics implementation will see interleaved requests
begin/end like:

begin request 1
begin request 2
end request 1
begin request 3
end request 2
....

**Comment by vietj** [comment]
also that would make code inside the application that calls the `routed` like the metrics incorrect, because the request end event would be emitted before the application calls the `routed` method.

**Comment by TFyre** [comment]
This should be fine, since the metric is tied to the request... and a new request is being created for every message in `handleMessage`

**Pull request merged**